rust/numaflow-core/src/config/components.rs

```diff
@@ -96,6 +97,51 @@ pub(crate) mod source {
         }
     }
 
+    impl TryFrom<Box<numaflow_models::models::ServingSource>> for SourceType {
+        type Error = Error;
+        // FIXME: Currently, the same settings comes from user-defined settings and env variables.
+        // We parse both, with user-defined values having higher precedence.
+        // There should be only one option (user-defined) to define the settings.
+        fn try_from(cfg: Box<numaflow_models::models::ServingSource>) -> Result<Self> {
+            let mut setting =
+                serving::Settings::load().map_err(|e| Error::Config(e.to_string()))?;
+            setting.tid_header = cfg.msg_id_header_key; // FIXME: check if this is correct
+
+            if let Some(auth) = cfg.auth {
+                if let Some(token) = auth.token {
+                    let secret = crate::shared::create_components::get_secret_from_volume(
+                        &token.name,
+                        &token.key,
+                    )
+                    .map_err(|e| Error::Config(format!("Reading API auth token secret: {e:?}")))?;
+                    setting.api_auth_token = Some(secret);
+                } else {
+                    tracing::warn!("Authentication token for Serving API is specified, but the secret is empty");
+                };
+            }
+
+            if let Some(ttl) = cfg.store.ttl {
+                if ttl.is_negative() {
+                    return Err(Error::Config(format!(
+                        "TTL value for Redis store can not be negative. Provided value = {ttl:?}"
```

Comment:
```suggestion
                        "TTL value for the store can not be negative. Provided value = {ttl:?}"
```

removed the word `Redis` because `DDB` night come too.